Lib/site.py

```diff
@@ -390,6 +390,23 @@ def setcopyright():
         files, dirs)
 
 
+def _register_detect_pip_usage_in_repl():
+    old_excepthook = sys.excepthook
+
+    def detect_pip_usage_in_repl(typ, value, traceback):
+        if typ is SyntaxError and (
+            "pip install" in value.text or "pip3 install" in value.text
+        ):
+            value.msg = ("The Python package manager (pip) can only be used"
```

Comment:
Since there is a chance of false detection, it is worth to include the original message in the result.

Reply:
The original message is just `"invalid syntax"`, so to put that back in would require going from what we have now:

```
  File "<string>", line 1
    pip install qwerty
              ^
SyntaxError: The Python package manager (pip) can only be used from outside of Python.
Please try the `pip` command in a separate terminal or command prompt.
```

to something like the more expansive:

```
  File "<string>", line 1
    pip install qwerty
              ^
SyntaxError: invalid syntax

Note: The Python package manager (pip) can only be used from outside of Python.
Please try the `pip` command in a separate terminal or command prompt.
```

I'm happy with either.

Reply:
It is not always "invalid syntax".
```
>>>     x = 'pip install'
  File "<stdin>", line 1
    x = 'pip install'
    ^
IndentationError: unexpected indent
```

Reply:
I think this is why [the original suggestion](https://bugs.python.org/msg276506) from ncoghlan has `is SyntaxError` rather than `isinstance`, so subclasses (`IndentationError` & `TabError`) don't get caught.